### PR TITLE
fix: delete Lassie temp when zinniad starts

### LINKED ISSUE #245: Delete Lassie temp dir when `zinniad` starts
See https://github.com/filecoin-station/roadmap/issues/19

On process start, delete all files in Lassie’s temp dir. These files may be left over if Station exits Zinnia while a CID retrieval request is in flight.

### PR DESCRIPTION
**Pull request by bajtos**
Resolves #245

### CODE REVIEW

**Comment by bajtos** [comment]
The new structure:

Screenshot 2023-06-21 at 17 10 38

File: daemon/main.rs [hunk 1]
```diff
@@ -104,3 +105,38 @@ fn exit_with_error(error: Error) {
...
+            lassie_temp_dir
+        )
+    })?;
+
+    for ent in lassie_files {
+        match ent {
+            Err(err) => log::warn!("Cannot parse dir entry in Lassie tempdir: {:?}", err),
+            Ok(f) => {
+                let p = f.path();
+                log::trace!("Removing Lassie temp file {:?}", p);
```
Comment (juliangruber) [inline daemon/main.rs:133]: I assume lassie doesn't place directories in here?

Comment (juliangruber) [reply daemon/main.rs:133]: If so, could you add this constraint as a comment please?

**Review by juliangruber — APPROVED** [review]
great work!

File: daemon/main.rs (hunk 1 above)
Comment (bajtos) [reply daemon/main.rs:133]: > I assume lassie doesn't place directories in here?

Yes, that's my assumption too.

> If so, could you add this constraint as a comment please?

Great idea!

**Pull request merged**